File: Python Code/eom.py
def eom (t,x):
################################################################################
## eom calculates the state derivative vectors at the given time and states
#
#  Author: Kshitij Mall
#
#  Modified: 18 February 2019
#  Parameters:
#
#    Input: double t, the given time.
#
#           double x, the given state vector.
#
#    Output: double xdot: the state derivative vector
#
################################################################################
# Import the necessary packages
    import numpy as np
    # Inputs to calculate the state derivatives
    g = 9.80665
    re = 6378000
    B = 157
    rho0 = 1.225
    H = 7200
    xdot = [x[1]*np.sin(x[2]),\
           -rho0*np.exp(-x[0]/H)*x[1]**2/(2*B)-g*np.sin(x[2]),\
           (x[1]/(re+x[0]) - g/x[1])*np.cos(x[2])] 
    return xdot # Calculate the state derivative vector

File: C++ Code/eom.cpp
#include <iostream>
#include <math.h>
using namespace std;

// function to generate and retrun random numbers.
double * eom(double t, double x[]) {

    const double g = 9.80665;
    const double re = 6378000.0;
    const double B = 157.0;
    const double rho0 = 1.225;
    const double H = 7200.0;
    static double xdot[3];

    xdot[0] = x[1]*sin(x[2]);
    xdot[1] = -rho0*exp(-x[0]/H)*pow(x[1],2.0)/(2*B)-g*sin(x[2]);
    xdot[2] = (x[1]/(re+x[0]) - g/x[1])*cos(x[2]);

    cout << xdot[0] << "\n" << xdot[1] << "\n" << xdot[2] << "\n";
    return xdot;
}
